dsgrid/config/date_time_dimension_config.py

```diff
@@ -50,8 +57,89 @@ def check_dataset_time_consistency(self, load_data_df):
                 f"load_data {time_col}s do not match expected times. mismatch={mismatch}"
             )
 
-    def convert_dataframe(self, df, project_time_dim):
-        # TODO: handle time zone
+    def get_time_dataframe(self):
+        time_col = self.get_timestamp_load_data_columns()
+        assert len(time_col) == 1, time_col
+        time_col = time_col[0]
+        schema = StructType([StructField(time_col, TimestampType(), False)])
+
+        if self.model.timezone in [TimeZone.LOCAL, TimeZone.NONE]:
+            # TODO: handle local time zone
+            schema = StructType(
+                [
+                    StructField(time_col, StringType(), False),
+                    StructField("timezone", StringType(), False),
+                ]
+            )
+            raise ValueError("TimeZone = LOCAL or NONE needs fixing")
+
+        model_time = self.list_expected_dataset_timestamps()
+        spark = SparkSession.builder.appName("dgrid").getOrCreate()
```

Comment:
Consider using dsgrid/utills/spark.py. The existing dsgrid _get_active_session could be made public. Consider that SparkSession.getActiveSession will never create a new session but getOrCreate might.